x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,529 @@
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+If your CA hasn't changed, you can just generate a new certificate, save it with
+the same name as your existing certificate, and replace the existing file.
+You'll need to generate new certificates and keys for both the transport layer
+and the HTTP layer.
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the certificate that is currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+Create a new PKCS#12 file that contains a certificate and key and replace
+the existing PKCS#12 file (such as `elastic-certificates.p12`) with the new
+certificate. 
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+security certificates that are currently in use. You'll use the same names
+for your new certificate.
++
+In this example, the keystore and truststore are pointing to different files.
+Your configuration might use the same file that contains the certificate and CA.
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.keystore.password: password
+xpack.security.transport.ssl.truststore.path: config/elastic-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+xpack.security.transport.ssl.truststore.password: password
+----
+
+. On any node in your cluster, generate a certificate and private key for your
+node. You must include the CA that was used to sign the certificate currently
+in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA file used to sign your certificates. The
+   default file name from the `elasticsearch-certutil` tool is
+   `elastic-stack-ca.p12`.
+====
++
+    a. Enter the password for your CA when prompted.
+
+    b. Enter the name for your certificate, which must match the name of the
+    certificate currently in use.
+
+    c. (Optional) Enter a password for your certificate. 
+
+. If you entered a password when creating the node certificate, run the following commands to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.truststore.secure_password
```

Comment:
Since this points to the CA keystore in our example, they should not overwrite/set its password, which presumably hasn't changed

Reply:
++ I'll remove the command for updating the truststore password, which isn't necessary here.